### PR TITLE
Feature/enhancements

### PR DESCRIPTION
**Pull request by abashmak**

### CODE REVIEW

File: src/app/content/services/backend.service.ts [hunk 1]
```diff
@@ -15,13 +15,17 @@ export interface IBackendService {
...
-const baseUrl = `${window.location.protocol}//${window.location.hostname}:${window.location.port}${environment.baseHref}`;
-
 @Injectable()
 export class BackendService implements IBackendService {
   private headers: Headers;
+  private baseUrl: String;
 
-  constructor(private http: AuthHttp) { }
+  constructor(private http: AuthHttp) {
+    this.baseUrl = `${window.location.href}`.replace('/content', '');
```
Comment (DmitryEfimenko) [inline src/app/content/services/backend.service.ts]: actually didn't mean to approve this PR. Thought it only removes unused imports, but then saw this, which I really think is a hacky solution. There should be a better way.

**Comment by DmitryEfimenko** [comment]
really don't like the hacky solution to get rid baseUrl variable. There should be a better way

**Comment by abashmak** [comment]
Arguably it's no more hacky then hardcoding baseHref in environment. This way however, the site can be published to any path without having to modify anything in the code. Suggest a better way of accomplishing that if you can think of one.

**Comment by DmitryEfimenko** [comment]
Just like you said, introducing a new route will break the code right away. That's not a sign of a well-written code.
I think you might be able to get the base href value by importing [`APP_BASE_HREF`](https://angular.io/docs/ts/latest/api/common/index/APP_BASE_HREF-let.html) from `@angular/common`
Import it, and see what value it has. It might be exactly what you need.

**Comment by abashmak** [comment]
APP_BASE_HREF is an input to the application, not a reference from the framework. I was looking at [Location](https://angular.io/docs/ts/latest/api/common/index/Location-class.html), but it seems to be only for components, can't inject it into a service.

**Comment by abashmak** [comment]
Ok, I think I found a more elegant solution using Location. Seems to work.

**Comment by DmitryEfimenko** [comment]
This is perfect! Seems like this is exactly how it should be done.

**Pull request merged**